### PR TITLE
Fix "a15 cannot be used in asm here" on ESP8266 + 32x32 code.

### PR DESCRIPTION
**Pull request by marcmerlin**
Merge my backlog of updates to the demo code, including the most recent ESP8266 compile fix.

Highlights:
- Adds more matrix ordering examples and 32x32 support
- Adds demo code to make sure matrices are tiled in the correct order (count_pixels)
- ￼Fix "a15 cannot be used in asm here" on ESP8266 
https://github.com/adafruit/Adafruit_NeoMatrix/issues/18

Tests:
Tested on ESP8266 in both 16x16 and 32x32 resolutions.

### CODE REVIEW

**Comment by marcmerlin** [comment]
@ladyada  any idea about this failure? Not sure what's wrong.
```
################################################################################
SWITCHING TO adafruit:nrf52:feather52840:softdevice=s140v6,debug=l0
Installing adafruit:nrf52 ✓
################################################################################
	MatrixGFXDemo.ino ❌
```

**Comment by ladyada** [comment]
rebase from `master` to get the CI fix

**Comment by marcmerlin** [comment]
thanks for the help with CI fix.

**Comment by marcmerlin** [comment]
Oh crap, looks like I missed this up. Let me try again.

**Comment by marcmerlin** [comment]
@ladyada  clang formatter is getting a bit picky :) 
What's the magic command I'm supposed to run to format the way it wants on my side? (I looked for a mention of it in the PR guide text, but didn't find anything)

**Comment by ladyada** [comment]
clang-format -i filename

**Comment by marcmerlin** [comment]
Ok, sorry that it took several tries, but it's ready for you finally @ladyada

**Review by ladyada — APPROVED** [review]

**Comment by ladyada** [comment]
nice!
